src/test/java/teammates/test/cases/search/InstructorSearchTest.java

```diff
@@ -0,0 +1,90 @@
+package teammates.test.cases.search;
+
+import java.util.Arrays;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.InstructorSearchResultBundle;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.storage.api.InstructorsDb;
+import teammates.storage.search.InstructorSearchDocument;
+import teammates.storage.search.InstructorSearchQuery;
+import teammates.test.driver.AssertHelper;
+
+/**
+ * SUT: {@link InstructorsDb}, {@link InstructorSearchDocument},
+ * {@link InstructorSearchQuery}.
+ */
+public class InstructorSearchTest extends BaseSearchTest {
+    @Test
+    public void allTests() {
+        InstructorsDb instructorsDb = new InstructorsDb();
+
+        InstructorAttributes ins1InCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        InstructorAttributes ins2InCourse1 = dataBundle.instructors.get("instructor2OfCourse1");
+        InstructorAttributes ins1InCourse2 = dataBundle.instructors.get("instructor1OfCourse2");
+        InstructorAttributes ins2InCourse2 = dataBundle.instructors.get("instructor2OfCourse2");
+        InstructorAttributes insInArchivedCourse = dataBundle.instructors.get("instructorOfArchivedCourse");
+        InstructorAttributes insInUnregCourse = dataBundle.instructors.get("instructor5");
+
+        ______TS("success: search for instructors in whole system; query string does not match anyone");
+
+        InstructorSearchResultBundle results =
+                instructorsDb.searchInstructorsInWholeSystem("non-existent");
```

Comment:
Suggest one more test case.
`""`, empty string for the search keywords. Our system will return `nothing` for empty search keywords.